cmake/FindESMF.cmake

```diff
@@ -109,6 +109,9 @@ if(EXISTS ${ESMFMKFILE})
     endif()
   endif()
 
+  # Add target alias to facilitate unambiguous linking
+  add_library(ESMF::ESMF ALIAS ESMF)
```

Comment:
fails if find_package(ESMF) is called multiple times. This is likely in the layered build schemes used to build coupled applications.

Recommended change
```
    if(NOT TARGET ESMF::ESMF)
      add_library(ESMF::ESMF ALIAS ESMF)
    endif()
```